corehq/apps/builds/management/commands/add_commcare_build.py

```diff
@@ -1,26 +1,91 @@
-from django.core.management.base import BaseCommand, CommandError
+import random
+from github import Github
 
-from corehq.apps.builds.models import CommCareBuild
+from django.core.management.base import BaseCommand, CommandError
+from corehq.apps.builds.models import CommCareBuild, CommCareBuildConfig, BuildMenuItem, BuildSpec
 
 
 class Command(BaseCommand):
     help = ('Adds a commcare build, labeled with the version (x.y.z) and build_number (an incrementing integer)\n'
             'to get started see https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/builds/README.md')
 
     def add_arguments(self, parser):
-        parser.add_argument('build_path')
-        parser.add_argument('version')
-        parser.add_argument('build_number')
+        parser.add_argument('build_path', nargs='?')
+        parser.add_argument('version', nargs='?')
+        parser.add_argument('build_number', nargs='?')
+        parser.add_argument(
+            '-l',
+            '--latest',
+            action='store_true',
+            help="add the latest remote CommCare build version"
+        )
 
     def handle(self, build_path, version, build_number, **options):
-        try:
-            build_number = int(build_number)
-        except ValueError:
-            raise CommandError("Build Number %r is not an integer" % build_number)
-
-        try:
-            CommCareBuild.create_from_zip(build_path, version, build_number)
-        except Exception as e:
-            raise CommandError("%s" % e)
-        self.stdout.write('Build %s #%s created\n' % (version, build_number))
-        self.stdout.write('You can see a list of builds at [your-server]/builds/\n')
+        if options.get('latest'):
+            _create_build_from_latest_remote_version()
+        else:
+            if build_path and version and build_number:
+                try:
+                    build_number = int(build_number)
+                except ValueError:
+                    raise CommandError("Build Number %r is not an integer" % build_number)
+
+                try:
+                    CommCareBuild.create_from_zip(build_path, version, build_number)
+                except Exception as e:
+                    raise CommandError("%s" % e)
+                self.stdout.write('Build %s #%s created\n' % (version, build_number))
+                self.stdout.write('You can see a list of builds at [your-server]/builds/\n')
+            else:
+                raise CommandError("<build_path>, <version> or <build_number> not specified!")
+
+
+def _create_build_from_latest_remote_version():
+    version = _get_latest_commcare_build_version()
+    # arbitrary build_number
+    build_number = random.randint(0, 100)
```

Comment:
100 is low enough we might hit duplicates relatively soon - there's a new build every month or two. How about a bigger number, or seconds since the epoch (although that's long)?

Reply:
Also I don't think duplicates matter - why don't we just always use say, `0`, so it'll be especially easy to identify whether the number is real or arbitrary?

Reply:
Actually, looking at this all a littler closer, `build_number` isn't a required field, so I think it'd be preferable to just omit it.

Reply:
Yeah, I'm going to omit it. Rather leave it as it was. Although, it did break if no build_number was given due to the fact that we fetch it from the request (`request.POST.get('build_number')`), so if no build_number was given the value would just be an empty string which is _not_ accepted by the model rather than None which is. I would just add a something like `request.POST.get('build_number') or None` with a comment as to why.